game-backend/src/game-engine/intent/MoveIntent.ts

```diff
@@ -0,0 +1,50 @@
+import { Intent } from './Intent';
+import { State } from '../../../../darwin-types/State';
+import { UserContext } from '../../../../darwin-types/UserContext';
+
+export enum Direction {
+  Up = 'UP',
+  Down = 'DOWN',
+  Left = 'LEFT',
+  Right = 'RIGHT',
+}
+
+/**
+ * Represents the intent to move the player around.
+ */
+export default class MoveIntent implements Intent {
+  constructor(private direction: Direction) {}
+
+  execute(state: State, userContext: UserContext): State {
+    const unit = state.objectMap[userContext.unitId];
+
+    const position = {
+      ...unit.position,
+    };
+    switch (this.direction) {
+      case Direction.Up:
+        position.x--;
```

Comment:
Consider using the x parameter as the vertical axis.

Reply:
Let's handle these kind of discussions as part of the actual implementation. Atm it's really only about the engine :v: